batchflow/models/eager_torch/layers/core.py

```diff
@@ -0,0 +1,176 @@
+""" Basic Torch layers. """
+import inspect
+
+import numpy as np
+import torch
+import torch.nn as nn
+
+from ..utils import get_shape, get_num_channels, get_num_dims
+
+
+
+class Flatten(nn.Module):
+    """ A module which reshapes inputs into 2-dimension (batch_items, features). """
+    def forward(self, x):
+        return x.view(x.size(0), -1)
+
+
+
+class Dense(nn.Module):
+    """ Dense layer. """
+    def __init__(self, units=None, out_features=None, bias=True, inputs=None):
+        #pylint: disable=eval-used
+        super().__init__()
+
+        in_units = np.prod(get_shape(inputs)[1:])
+        units = units or out_features
+        if isinstance(units, str):
+            units = eval(units, {}, {key: in_units for key in ['S', 'same']})
```

Comment:
Please extract eval to a function and add proper comments

Reply:
Done